Tasks/KubernetesV1/src/kubernetes.ts

```diff
@@ -82,14 +99,65 @@ function executeKubectlCommand(clusterConnection: ClusterConnection, command: st
         "TaskEndpointId",
         "KubernetesV1",
         JSON.stringify(telemetry));
-    var result = "";
-    return commandImplementation.run(clusterConnection, command, (data) => result += data)
-    .fin(function cleanup() {
-        var commandOutputLength = result.length;
-        if (commandOutputLength > environmentVariableMaximumSize) {
-            tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
-        } else {
-            tl.setVariable('KubectlOutput', result);
-        }
-    });
+
+    // The output result can contain more than one Json objects
+    // We want to parse each of the objects separately, hence push the output in JSON array form    
+    var result = [];
+    return commandImplementation.run(clusterConnection, command, (data) => result.push(data))
+        .fin(function cleanup() {
+            console.log("commandOutput" + result);
+            var commandOutputLength = result.length;
+            if (commandOutputLength > environmentVariableMaximumSize) {
+                tl.warning(tl.loc("OutputVariableDataSizeExceeded", commandOutputLength, environmentVariableMaximumSize));
+            } else {
+                tl.setVariable('KubectlOutput', result.toString());
+            }
+
+            // For each output, check if it contains a JSON object
+            result.forEach(res => {
+                if (IsJsonString(res)) {
```

Comment:
What if the output was set to `yaml`

Reply:
Added checks for parsing data for both yaml and json formats.